apis/v1alpha2/validation/validation_test.go

```diff
@@ -424,3 +426,68 @@ func portNumberPtr(p int) *gatewayv1a2.PortNumber {
 	result := gatewayv1a2.PortNumber(p)
 	return &result
 }
+
+func TestValidateGatewayClassUpdate(t *testing.T) {
+	type args struct {
+		oldClass *gatewayv1a2.GatewayClass
+		newClass *gatewayv1a2.GatewayClass
+	}
+	tests := []struct {
+		name string
+		args args
```

Comment:
This is a tiny nit/question - any reason not to just directly embed oldClass and newClass here? At first I thought this might be the specific type that `ValidateGatewayClassUpdate` takes, but it looks like it just takes 2 individual GatewayClasses. This doesn't have to block this PR though.

Reply:
I use https://github.com/cweill/gotests to auto-generate these tables.